e2e/infra/pod_state.go

```diff
@@ -30,7 +25,30 @@ func AllPodsReady(ns string) func() bool {
 	}
 }
 
-func podStatus(pod, ns, status string) bool {
+// GetAllPods returns names of all pods from a given namespace
+func GetAllPods(ns string) []string {
+	podsCmd := shell.ExecuteInDir(".",
+		"oc", "get", "pod",
+		"-n", ns,
+		"-o", "jsonpath={.items[*].metadata.name}")
+	<-podsCmd.Done()
+	return strings.Split(strings.Trim(fmt.Sprintf("%s", podsCmd.Status().Stdout), "[]"), " ")
+}
+
+// StateOf returns state of the pod
+func StateOf(ns, pod string) {
+	state := shell.Execute("oc get pod " + pod + " -n " + ns + " -o yaml")
```

Comment:
Did we conclude on the oc vs kubectrl?

Reply:
I was thinking about this one, but I am not sure how to break it down. This is not only about using `kubectl` but also having tests ran against vanilla k8s with (presumably) vanilla istio. This would require not only changing CLI tools invocations but also assumptions about the cluster (e.g. we do create some maistra-specific CR right now).

We could have one task switching from `oc` to `kubectl` at first and then the `k8s` one as separated step, but I am not sure if split would bring any value. WDYT?

Reply:
If our goal is to have `ike` working with any `k8s` distro then we should probably go for it as preps on what else should we change in the production code too.

Reply:
We can start by writing down some basics in an Epic or something and split out tasks from there.